lib/HALInterfaces/src/INetwork.h

```diff
@@ -89,15 +89,21 @@ class INetwork
      * Set client configuration.
      *
      * @param[in] clientId      Client ID.
+     * @param[in] ssid          SSID of the WiFi network.
+     * @param[in] password      Password of the WiFi network.
      * @param[in] brokerAddress Broker address to connect to.
      * @param[in] brokerPort    Broker port to connect to.
+     * @param[in] birthTopic    Birth topic. If empty, no birth message is used.
+     * @param[in] birthMessage  Birth message.
      * @param[in] willTopic     Last will topic. If empty, no last will is used.
      * @param[in] willMessage   Last will message.
      * @param[in] reconnect     If true, the client will try to reconnect to the broker, if the connection is lost.
      * @return If successfully set, returns true. Otherwise, false.
      */
-    virtual bool setConfig(const String& clientId, const String& brokerAddress, uint16_t brokerPort,
-                           const String& willTopic, const String& willMessage, bool reconnect) = 0;
+    virtual bool setConfig(const String& clientId, const String& ssid, const String& password,

```

Comment:
We should think about using a struct.

Reply:
You are right. I would make this a new issue and resolve it later.